### PR TITLE
Add mangrove tree type

### LINKED ISSUE #2197: //brush mangrove
### What feature do you want to see added?

Currently, the tree brush does not support mangrove trees. 

I would like this to support mangrove trees.

### Are there any alternatives?

Not that I know of, I checked other issues I couldn't find any with the content "mangrove". 

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by TheMeinerLP**
## Overview
Fixes #2197 

## Description
Add some debug messages for exceptions. Also add mangrove as a tree type

## Submitter Checklist

- [X] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [X] Ensure that the pull request title represents the desired changelog entry.
- [X] New public fields and methods are annotated with `@since TODO`.
- [X] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NotMyFault** [comment]
Tests are failing.

**Review by SirYwell — CHANGES_REQUESTED** [review]
Catching unchecked exceptions should be a last resort workaround if we can't control the code. In this case, however, it looks like we can properly deal with it instead of catching exceptions?

**Comment by TheMeinerLP** [comment]
> Catching unchecked exceptions should be a last resort workaround if we can't control the code. In this case, however, it looks like we can properly deal with it instead of catching exceptions?

There is currently no other way for handle thats type. 
I tried to check it before we register that tree type. But everytime it fails.
Now this solution is the friendliest one for every user of fawe.

**Comment by SirYwell** [comment]
The NPE catches can clearly be replaced with null checks earlier. Registering the mapping can be guarded by a version check.

**Comment by TheMeinerLP** [comment]
> The NPE catches can clearly be replaced with null checks earlier. Registering the mapping can be guarded by a version check.

The NPE comes from Bukkit it self. We can't handle it simple null checks

**Comment by SirYwell** [comment]
I assume the NPE comes from `TreeType bukkitType = BukkitWorld.toBukkitTreeType(treeType);` being null? If so, we can simply return directly there

**Comment by TheMeinerLP** [comment]
> I assume the NPE comes from `TreeType bukkitType = BukkitWorld.toBukkitTreeType(treeType);` being null? If so, we can simply return directly there

The NPE comes from internal of Bukkit not from our code. `bukkitWorld.generateTree(<PLACEHOLDER>, type)` throws a exception when we give over the tree type

**Comment by SirYwell** [comment]
Which type? It will be null if the type doesn't exist on that version -> NPE

**Comment by TheMeinerLP** [comment]
> Which type? It will be null if the type doesn't exist on that version -> NPE

I will try to handle it with null check and add a comment why.
Also i will add the debug again

**Review by NotMyFault — COMMENTED** [review]
This should really be in upstream.

**Comment by TheMeinerLP** [comment]
I can do that 😄

File: worldedit-core/src/main/java/com/sk89q/worldedit/util/TreeGenerator.java [hunk 1]
```diff
@@ -49,6 +49,7 @@ public enum TreeType {
         REDWOOD("Spruce tree", "spruce", "redwood", "sequoia", "sequoioideae"),
         TALL_REDWOOD("Tall spruce tree", "tallspruce", "bigspruce", "tallredwood", "tallsequoia", "tallsequoioideae"),
         MEGA_REDWOOD("Large spruce tree", "largespruce", "megaredwood"),
+        MANGROVE("Mangrove", "mangrove"),
```
Comment (MattBDev) [inline worldedit-core/src/main/java/com/sk89q/worldedit/util/TreeGenerator.java:52]:
```suggestion
        MANGROVE("Mangrove tree", "mangrove"),
```

**Review by MattBDev — CHANGES_REQUESTED** [review]
I am requesting some changes since this will be a commit will have to PR to EngineHub/WorldEdit. 
In addition to the recommendations I commented on, I would also like to see Tall Mangrove tree's added to this PR before I approve.

**Comment by NotMyFault** [comment]
Upstreamed by dordsor, due to lack of action by the PR creator.

**Pull request closed without merging**